modules/library/referencing/src/main/java/org/geotools/measure/Units.java

```diff
@@ -115,24 +115,70 @@ public static void registerCustomUnits(SimpleUnitFormat format) {
 
         format.label(Units.PPM, "ppm");
     }
+    /**
+     * Detects and auto-corrects units defined with a multiplication factor that is close, but not exactly, as expected.
+     * <p>
+     * This is used to handle {@link NonSI#DEGREE_ANGLE} and {@link USCustomary#FOOT_SURVEY} matches.
+     * <p>
+     * @param unit 
+     * @return Unit, possibly auto-corrected based on multiplication factor match
+     */
+    public static Unit<?> autoCorrect(Unit<?> unit) {
```

Comment:
This is they key method, identifying common multiplication factors, and providing the built-in unit as a replacement.